lti_consumer/lti_1p3/consumer.py

```diff
@@ -704,3 +708,178 @@ def enable_nrps(self, context_memberships_url):
+        launch_data = get_data_from_cache(lti_message_hint)
+
+        if launch_data.message_type == "LtiStartProctoring":
+            proctoring_claims = self.get_start_proctoring_claims()
+        elif launch_data.message_type == "LtiEndAssessment":
+            proctoring_claims = self.get_end_assessment_claims()
+        else:
+            raise ValueError('lti_message_hint must \"LtiStartProctoring\" or \"LtiEndAssessment\".')
+
+        self.set_extra_claim(proctoring_claims)
+
+        return super().generate_launch_request(preflight_response)
+
+    def check_and_decode_token(self, token):
+        """
+        Decodes a Tool JWT token and validates OAuth specific claims. Validates that required claims are provided but
+        does not validate their values. Use validate_token_contents to validate values. Returns a dictionary
+        representation of key proctoring claims in the Tool JWT token.
+
+        Arguments:
+            * token (string): a JWT
+        """
+        # Decode token and check expiration.
+        proctoring_response = self.tool_jwt.validate_and_decode(token)
+
+        claim_key = "https://purl.imsglobal.org/spec/lti-ap/claim/session_data"
+        session_data = proctoring_response.get(claim_key)
+        if session_data is None:
+            raise MissingRequiredClaim(f"Token is missing required {claim_key} claim.")
+
+        claim_key = "https://purl.imsglobal.org/spec/lti/claim/resource_link"
+        resource_link = proctoring_response.get(claim_key)
+        if resource_link is None or not resource_link:
+            raise MissingRequiredClaim(f"Token is missing required {claim_key} claim.")
+
+        response = {
+            'end_assessment_return': proctoring_response.get(
+                "https://purl.imsglobal.org/spec/lti-ap/claim/end_assessment_return",
+            ),
+            'verified_user': proctoring_response.get("https://purl.imsglobal.org/spec/lti-ap/claim/verified_user", {}),
+            'resource_link': proctoring_response["https://purl.imsglobal.org/spec/lti/claim/resource_link"],
+            'session_data': proctoring_response["https://purl.imsglobal.org/spec/lti-ap/claim/session_data"],
+            'attempt_number': proctoring_response.get("https://purl.imsglobal.org/spec/lti-ap/claim/attempt_number")
+        }
+
+        return response
+
+    def validate_token_contents(self, token_contents):
+        """
+        Validates the contents of a Tool JWT token. Raises InvalidClaimValue appropriately.
+
+        Arguments:
+            * token_contents (dict): a dictionary representation of the Tool JWT claims, as returned by
+              check_and_decode_token.
+        """
+        # TODO: We only HAVE to validate the session_data claim. Other claims must be validated, but they should be
+        #       validated by the ToolKeyHandler class.
+        # TODO: An Assessment Platform MAY reject the Start Assessment message if a required identity claim is missing
```

Comment:
this sounds like something we can revisit once we start to look at integration with name affirmation / verification. It may make sense to be less restrictive at the start.